### PR TITLE
Move loading of the intiial screen in `TheActivity` to the Mobius loop

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/2342/move-loading-of-the-initial-screen-in-theactivity-to-the-mobius-loop

### CODE REVIEW

**Comment by vinaysshenoy** [comment]
@msasikanth Sorry for the slightly largish PR, but this was hard to do without breaking the navigation flow in between commits. So I decided to make it part of one changeset.

File: app/src/main/java/org/simple/clinic/forgotpin/createnewpin/ForgotPinCreateNewPinScreenKey.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ import org.simple.clinic.R
 import org.simple.clinic.router.screen.FullScreenKey
 
 @Parcelize
-class ForgotPinCreateNewPinScreenKey : FullScreenKey {
+object ForgotPinCreateNewPinScreenKey : FullScreenKey {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/forgotpin/createnewpin/ForgotPinCreateNewPinScreenKey.kt:9]: Same comment as before, we can convert this to a data class.

**Review by msasikanth — COMMENTED** [review]
Looks good overall. Left couple of comments PTAL.

File: app/src/main/java/org/simple/clinic/main/TheActivityUpdate.kt [hunk 2]
```diff
@@ -51,15 +56,15 @@ class TheActivityUpdate : Update<TheActivityModel, TheActivityEvent, TheActivity
       LoggedInStatus.LOGGED_IN, LoggedInStatus.OTP_REQUESTED, LoggedInStatus.RESET_PIN_REQUESTED, LoggedInStatus.UNAUTHORIZED -> true
     }
 
-    val initialScreenEffect = when {
-      shouldShowAppLockScreen -> ShowAppLockScreen
-      userDisapproved -> ShowAccessDeniedScreen
-      canMoveToHomeScreen && !userDisapproved -> ShowHomeScreen
-      user.loggedInStatus == LoggedInStatus.RESETTING_PIN -> ShowForgotPinScreen
+    val initialScreen = when {
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/main/TheActivityUpdate.kt:63]: Yeah, I couldn't think of a better way to do this right now. My priority was to remove the database call for the user we were making in `TheActvity#onCreate`, so I feel this is acceptable for now. Once this merged in, we could look at better ways to do this, I think.

File: app/src/main/java/org/simple/clinic/login/applock/AppLockScreenKey.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@ import kotlinx.parcelize.Parcelize
 import org.simple.clinic.navigation.v2.ScreenKey
 
 @Parcelize
-object AppLockScreenKey : ScreenKey(), Parcelable {
+class AppLockScreenKey : ScreenKey(), Parcelable {
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/login/applock/AppLockScreenKey.kt]: This was made a `data class` in a later commit. Good point about moving the `analyticsName` to the constructor declaration though. Will make this change.

File: app/src/main/java/org/simple/clinic/forgotpin/createnewpin/ForgotPinCreateNewPinScreenKey.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/forgotpin/createnewpin/ForgotPinCreateNewPinScreenKey.kt:9]: Will make this change.

File: app/src/main/java/org/simple/clinic/main/TheActivityUpdate.kt [hunk 4]
```diff
@@ -43,8 +44,20 @@ class TheActivityUpdate : Update<TheActivityModel, TheActivityEvent, TheActivity
         .orElse(true) // Handle the case where the app is opened after a cold start
 
     val shouldShowAppLockScreen = shouldShowAppLockScreenForUser(user) && hasAppLockTimerExpired
+    val userDisapproved = user.status == UserStatus.DisapprovedForSyncing
 
-    return if (shouldShowAppLockScreen) dispatch(ShowAppLockScreen) else dispatch(ClearLockAfterTimestamp)
+    val canMoveToHomeScreen = when (user.loggedInStatus) {
+      LoggedInStatus.RESETTING_PIN -> false
+      LoggedInStatus.LOGGED_IN, LoggedInStatus.OTP_REQUESTED, LoggedInStatus.RESET_PIN_REQUESTED, LoggedInStatus.UNAUTHORIZED -> true
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/main/TheActivityUpdate.kt]: Hmm, maybe we can decide how to we want to format these and make it part of auto-format checks.

**Review by msasikanth — APPROVED** [review]

**Pull request merged**